File: app/src/main/java/com/brianmk/exposurenotes/FrameArrayAdapter.java
package com.brianmk.exposurenotes;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v4.content.ContextCompat;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import java.util.List;
import java.util.Locale;

public class FrameArrayAdapter extends ArrayAdapter<FrameData> {
    private static final String LOG_TAG = FrameArrayAdapter.class.getSimpleName();

    FrameArrayAdapter(Context context, List<FrameData> frameData) {
        super(context, 0, frameData);
    }

    @Override
    public @NonNull View getView(int pos, View view, @NonNull ViewGroup parent) {
        FrameData frame = getItem(pos);

        if (view == null) {
            view = LayoutInflater.from(getContext()).inflate(R.layout.frame_list_item,
                    parent,false);
        }

        // Populate the list view with information
        if (frame != null) {
            TextView frameNumberView = view.findViewById(R.id.number);
            if ((frame.getShutterIdx() > 0) && (frame.getApertureIdx() > 0)) {
                frameNumberView.setTextColor(ContextCompat.getColor(getContext(), R.color.dark_grey));
                ((TextView) view.findViewById(R.id.number)).setText(String.format(Locale.getDefault(), "%d", pos + 1));
            } else {
                frameNumberView.setTextColor(ContextCompat.getColor(getContext(), R.color.lighter_grey));

                ((TextView) view.findViewById(R.id.number)).setText(String.format(Locale.getDefault(), "%d", pos + 1));
            }

            ((TextView) view.findViewById(R.id.shutter)).setText(frame.getShutter());
            ((TextView) view.findViewById(R.id.aperture)).setText(frame.getAperture());
            ((TextView) view.findViewById(R.id.notes)).setText(frame.getNotes());


        } else {
            Log.d(LOG_TAG, "frame #" + pos + " is null.");
        }

        return view;
    }

}

File: app/src/main/java/com/brianmk/exposurenotes/FrameArrayAdapter.kt
package com.brianmk.exposurenotes

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import androidx.core.content.ContextCompat
import java.util.*

class FrameArrayAdapter internal constructor(context: Context, frameData: List<FrameData>) : ArrayAdapter<FrameData>(context, 0, frameData) {

    override fun getView(pos: Int, view: View?, parent: ViewGroup): View {
        var view = view
        val frame = getItem(pos)

        if (view == null) {
            view = LayoutInflater.from(context).inflate(R.layout.frame_list_item,
                    parent, false)
        }

        // Populate the list view with information
        if (frame != null) {
            val frameNumberView = view!!.findViewById<View>(R.id.number) as TextView

            if (frame.shutterIdx > 0 && frame.apertureIdx > 0) {
                frameNumberView.setTextColor(ContextCompat.getColor(context, R.color.colorPrimaryDark))
                (view.findViewById<View>(R.id.number) as TextView).text = String.format(Locale.getDefault(), "%02d", pos + 1)
            } else {
                frameNumberView.setTextColor(ContextCompat.getColor(context, R.color.light_grey))

                (view.findViewById<View>(R.id.number) as TextView).text = String.format(Locale.getDefault(), "%02d", pos + 1)
            }

            val shutterView = view.findViewById<View>(R.id.shutter) as TextView
            when (frame.shutter) {
                "B" -> shutterView.setTextColor(ContextCompat.getColor(context, R.color.dark_yellow))
                "T" -> shutterView.setTextColor(ContextCompat.getColor(context, R.color.blue))
                else -> shutterView.setTextColor(ContextCompat.getColor(context, R.color.dark_grey))
            }
            shutterView.text = frame.shutter

            (view.findViewById<View>(R.id.aperture) as TextView).text = frame.aperture
            (view.findViewById<View>(R.id.notes) as TextView).text = frame.notes

        } else {
            Log.d(LOG_TAG, "frame #$pos is null!?")
        }

        return view!!
    }

    companion object {
        private val LOG_TAG = FrameArrayAdapter::class.java.simpleName
    }

}
